code/addons/onboarding/src/App.tsx

```diff
@@ -0,0 +1,155 @@
+      step: '1:Welcome',
+      type: 'telemetry',
+    });
+  }, []);
+
+  useEffect(() => {
+    if (step !== '1:Welcome') {
+      api.emit(STORYBOOK_ADDON_ONBOARDING_CHANNEL, {
+        step,
+        type: 'telemetry',
+      });
+    }
+  }, [api, step]);
+
+  useEffect(() => {
+    let stepTimeout: number;
+    if (step === '4:VisitNewStory') {
+      setShowConfetti(true);
+      stepTimeout = window.setTimeout(() => {
+        setStep('5:ConfigureYourProject');
+      }, 2000);
+    }
+
+    return () => {
+      clearTimeout(stepTimeout);
+    };
+  }, [step]);
+
+  useEffect(() => {
+    const storyId = api.getCurrentStoryData()?.id;
+    api.setQueryParams({ onboarding: 'true' });
+    // make sure the initial state is set correctly:
+    // 1. Selected story is primary button
+    // 2. The addon panel is opened, in the bottom and the controls tab is selected
+    if (storyId !== 'example-button--primary') {
+      try {
+        api.selectStory('example-button--primary', undefined, {
+          ref: undefined,
+        });
+      } catch (e) {
+        //
+      }
+    }
+  }, []);
+
+  if (!enabled) {
+    return null;
+  }
+
+  return (
+    <ThemeProvider theme={theme}>
+      {enabled && showConfetti && (
+        <Confetti
+          numberOfPieces={800}
+          recycle={false}
+          tweenDuration={20000}
+          onConfettiComplete={(confetti) => {
+            confetti?.reset();
+            setShowConfetti(false);
+          }}
+        />
+      )}
+      {enabled && step === '1:Welcome' && (
+        <WelcomeModal
+          onProceed={() => {
+            setStep('2:StorybookTour');
+          }}
+          skipOnboarding={() => {
+            skipOnboarding();
+
+            api.emit(STORYBOOK_ADDON_ONBOARDING_CHANNEL, {
+              step: 'X:SkippedOnboarding',
+              where: 'WelcomeModal',
+              type: 'telemetry',
+            });
+          }}
+        />
+      )}
+      {enabled && (step === '2:StorybookTour' || step === '5:ConfigureYourProject') && (
+        <GuidedTour
+          api={api}
+          isFinalStep={step === '5:ConfigureYourProject'}
+          onFirstTourDone={() => {
+            setStep('3:WriteYourStory');
+          }}
+          codeSnippets={codeSnippets || undefined}
+          onLastTourDone={() => {
+            try {
+              api.selectStory('configure-your-project--docs');
+            } catch (e) {
+              //
+            }
+            api.emit(STORYBOOK_ADDON_ONBOARDING_CHANNEL, {
+              step: '6:FinishedOnboarding',
+              type: 'telemetry',
+            });
+            skipOnboarding();
+          }}
+        />
+      )}
+      {enabled && step === '3:WriteYourStory' && codeSnippets && (
+        <WriteStoriesModal
+          api={api}
+          codeSnippets={codeSnippets}
+          addonsStore={addons}
+          onFinish={() => {
+            api.selectStory('example-button--primary');
```

Comment:
```suggestion
            api.selectStory('example-button--warning');
```

My suggestion is the original code, see here: https://github.com/storybookjs/addon-onboarding/blob/main/src/App.tsx#L138. I think it was mistakenly replaced during the migration.